azure-iot-device/azure/iot/device/common/pipeline/config.py

```diff
@@ -77,3 +82,23 @@ def _sanitize_cipher(cipher):
             raise TypeError("Invalid type for 'cipher'")
 
         return cipher
+
+    @staticmethod
+    def _rectify_keep_alive(keep_alive):
```

Comment:
>rectify [](start = 9, length = 7)

Let's call it "_santize_keep_alive" for consistency with the other, similar method

Reply:
If we raise an exception when the value is out of bounds, then we're not sanitizing it.  Maybe `_validate_keep_alive`?

---
In reply to: [461187069](https://github.com/Azure/azure-iot-sdk-python/pull/635#discussion_r461187069) [](ancestors = 461187069)

Reply:
Good point - I had thought we were doing some type conversions as well, but upon closer look we aren't - maybe we should

---
In reply to: [461197405](https://github.com/Azure/azure-iot-sdk-python/pull/635#discussion_r461197405) [](ancestors = 461197405,461187069)

Reply:
i am actually just logging errors and assigning correct values , i am not throwing errors as such.. i thought of validation in case of throwing errors but since i am not throwing errors i called it rectify...sanitize also can be used but i was not cleaning up value...i was actually correcting them...so if it is much larger use max , if it negative or zero  use default.

---
In reply to: [461198869](https://github.com/Azure/azure-iot-sdk-python/pull/635#discussion_r461198869) [](ancestors = 461198869,461197405,461187069)